syntax/scan.go

```diff
@@ -462,11 +473,18 @@ start:
 
 	// comment
 	if c == '#' {
+		sc.startToken(val)
```

Comment:
The scanner should know whether comments are required, and skip the new logic if so.